src/redisearch_rs/rqe_iterators/src/rqe_iterator.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use ffi::t_docId;
+use inverted_index::RSIndexResult;
+
+/// The result of an iterator read or skipping operation
+pub enum RQEIteratorStatus {
+    /// The iterator has found a valid result, in either `read()` or `skip_to()`
+    OK(RSIndexResult),
+
+    /// A `skip_to()` has found a result with a `doc_id` greater than the requested `doc_id`
+    NotFound(RSIndexResult),
+}
```

Comment:
```suggestion
/// The outcome of [`RQEIterator::skip_to`].
pub enum SkipToOutcome {
    /// The iterator has a valid entry for the requested `doc_id`.
    Found(RSIndexResult),

    /// The iterator doesn't have an entry for the requested `doc_id`, but there are entries with an id greater than the requested one.
    NotFound(RSIndexResult),
}
```

Since we now use this enum exclusively for `skip_to`, we can make the docs more precise.